### DISCUSSION TITLE
Prevent delay when hover ends when there is delay present in animate prop.

### DISCUSSION THREAD

**Discussion by Nisan-Shrestha** [motiondivision/motion#2649] [Q&A] [answered]
Im trying to do as follow:
```
at initial scale is 0
animate scale is 1.25 with delay of 2 second
onHover scale is 5 but with no delay
onhover end the scale should go back to 1.25 or 1 without delay
```
I tried this but there is 2 second delay when hover ends before scale down animation begins.
I looked aorund and this method works for framer-motion 2.1.4 but not since motion 3.0

how do i solve this?
the sample code is:

```js
import React from "react";
import { motion } from "framer-motion";
import "./styles.css";

const size = 200;
const radius = 40;
const color = "#0CF";

export default function App() {
  const variants = {
    start:{scale:0},
    active: { scale: 1.25, transition: { delay: 2 } },
    hovering: {
      scale: 5,
      transition: { delay: 0 },
    },
  };

  return (
    <div className="App">
      <motion.div
        style={{
          width: size,
          height: size,
          backgroundColor: color,
          borderRadius: radius,
        }}
        variants={variants}
        initial="start"
        animate={"active"}
        whileHover={"hovering"}
        transition={{}}
      />
    </div>
  );
}
```

**Comment by Qwilliams815** [comment, marked as answer]
Hey there,

I ran into this problem as well, and while I never found a legitimate fix I did find a simple workaround using framer-motion's [onAnimationComplete](https://www.framer.com/motion/component/###onanimationcomplete) callback.

The idea is to store your delay value in state and update it after the initial animation is complete.

Here is your code with the implemented changes and a [codesandbox](https://codesandbox.io/p/sandbox/framer-override-delay-2wzzp9?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm07kziy20006356muduvj470%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm07kziy20002356mf0qy1h0v%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm07kziy20003356mfgcm9rm2%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm07kziy20005356mlrln6v8k%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm07kziy20002356mf0qy1h0v%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm07kziy20001356msd24go7s%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%257D%255D%252C%2522id%2522%253A%2522cm07kziy20002356mf0qy1h0v%2522%252C%2522activeTabId%2522%253A%2522cm07kziy20001356msd24go7s%2522%257D%252C%2522cm07kziy20005356mlrln6v8k%2522%253A%257B%2522id%2522%253A%2522cm07kziy20005356mlrln6v8k%2522%252C%2522activeTabId%2522%253A%2522cm07kztmj000r356mol4muuzm%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm07kziy20004356mk1eqdx40%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522cm07kztmj000r356mol4muuzm%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522cm07kziy20003356mfgcm9rm2%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cm07kziy20003356mfgcm9rm2%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) demo running framer-motion ^3.30:

```js
import React, { useState } from "react";
import { motion } from "framer-motion";
import "./styles.css";

const size = 200;
const radius = 40;
const color = "#0CF";

export default function App() {
  const [delay, setDelay] = useState(2); // Initial value

  const variants = {
    start: { scale: 0 },
    active: { scale: 1.25, transition: { delay: delay } },
    hovering: {
      scale: 5,
      transition: { delay: delay },
    },
  };

  return (
    <div className="App">
      <motion.div
        style={{
          width: size,
          height: size,
          backgroundColor: color,
          borderRadius: radius,
        }}
        variants={variants}
        initial="start"
        animate={"active"}
        whileHover={"hovering"}
        transition={{}}
        onAnimationComplete={() => setDelay(0)} // Subsequent value
      />
    </div>
  );
}
```
